BrowserKit/Sources/SiteImageView/ImageProcessing/BundleImageFetcher/BundleImageFetcher.swift

```diff
@@ -0,0 +1,139 @@
+            return [:]
+        }
+    }
+
+    private func decode(from data: Data) -> [String: FormattedBundledImage] {
+        let decoder = JSONDecoder()
+        var icons = [String: FormattedBundledImage]()
+
+        do {
+            let decodedImages = try decoder.decode([BundledImage].self, from: data)
+            for decodedImage in decodedImages {
+                guard let image = format(image: decodedImage) else {
+                    continue
+                }
+
+                icons[image.title] = image
+            }
+
+            if icons.isEmpty {
+                generalBundleError = ImageError.unableToGetFromBundle(.noBundleRetrieved("Bundle was empty"))
+            }
+
+            return icons
+
+        } catch {
+            let message = "Decoding BundledImage failed due to: \(error.localizedDescription.debugDescription)"
+            generalBundleError = ImageError.unableToGetFromBundle(.noBundleRetrieved(message))
+            return icons
+        }
+    }
+
+    private func format(image: BundledImage) -> FormattedBundledImage? {
+        let path = image.image_url.replacingOccurrences(of: ".png", with: "")
+        let title = image.title
+        let color = image.background_color
+        let filePath = bundleDataProvider.getPath(from: path)
+        guard let filePath = filePath else {
+            imagesErrors[title] = ImageError.unableToGetFromBundle(.imageFormatting("No filepath for image path: \(path)"))
+            return nil
+        }
+
+        var backgroundColor: UIColor
+        if color == "#fff" || color == "#FFF" {
+            backgroundColor = UIColor.clear
+        } else {
+            backgroundColor = UIColor(colorString: color.replacingOccurrences(of: "#", with: ""))
+        }
+
+        return FormattedBundledImage(backgroundColor: backgroundColor,
+                                     filePath: filePath,
+                                     title: title)
+    }
+}
+
+// MARK: - Extension UIImage
+private extension UIImage {
+    func withBackgroundAndPadding(color: UIColor, opaque: Bool = true) -> UIImage {
+        UIGraphicsBeginImageContextWithOptions(size, opaque, scale)
+
+        guard let ctx = UIGraphicsGetCurrentContext(), let image = cgImage else { return self }
+        defer { UIGraphicsEndImageContext() }
+
+        // Pad the image in a bit to make the favicons look better
+        let newSize = CGSize(width: size.width - 20, height: size.height - 20)
+        let rect = CGRect(origin: .zero, size: size)
+        let imageRect = CGRect(origin: CGPoint(x: 10, y: 10), size: newSize)
+        ctx.setFillColor(color.cgColor)
+        ctx.fill(rect)
+        ctx.concatenate(CGAffineTransform(a: 1, b: 0, c: 0, d: -1, tx: 0, ty: size.height))
+        ctx.draw(image, in: imageRect)
+
+        return UIGraphicsGetImageFromCurrentImageContext() ?? self
+    }
+}
```

Comment:
This doesn't need to be a UIImage extension, especially since it's private it can just be a helper method that takes the image as a parameter.